linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/CoreJSLib.scala

```diff
@@ -262,36 +268,64 @@ private[emitter] object CoreJSLib {
+                 *   Formal Proof of the Veltkamp/Dekker Algorithms
+                 *   https://dx.doi.org/10.1007/11814771_6
+                 * Section 3, with β = 2, t = 53, s = 53 - 24 = 29, x = av.
+                 * 53 is the number of effective mantissa bits in a Double;
+                 * 24 in a Float.
+                 *
+                 * ◦ is the round-to-nearest operation with a tie-breaking
+                 * rule (in our case, break-to-even).
+                 *
+                 *   Let C = βˢ + 1 = 536870913
+                 *   p = ◦(x × C)
+                 *   q = ◦(x − p)
+                 *   x₁ = ◦(p + q)
+                 *
+                 * Boldo proves that x₁ is the (t-s)-bit float closest to x,
+                 * using the same tie-breaking rule as ◦. Since (t-s) = 24,
+                 * this is the closest float32 (with 24 mantissa bits), and
+                 * therefore the correct result of `fround`.
+                 *
+                 * Boldo also proves that if the computation of x × C does not
+                 * cause overflow, then none of the following operations will
+                 * cause overflow. We know that x (av) is less than the
+                 * overflowThreshold, and overflowThreshold × C does not
+                 * overflow, so that computation can never cause an overflow.
+                 *
+                 * If the reader does not have access to Boldo's paper, they
+                 * may refer instead to
+                 *   Claude-Pierre Jeannerod, Jean-Michel Muller, Paul Zimmermann.
+                 *   On various ways to split a floating-point number.
+                 *   ARITH 2018 - 25th IEEE Symposium on Computer Arithmetic,
+                 *   Jun 2018, Amherst (MA), United States.
+                 *   pp.53-60, 10.1109/ARITH.2018.8464793. hal-01774587v2
+                 * available at
+                 *   https://hal.inria.fr/hal-01774587v2/document
+                 * Section III, although that paper defers some theorems and
+                 * proofs to Boldo's.
                  */
-                If(significand < 1, Block(
-                  twoPowE := twoPowE / 2,
-                  significand := significand * 2
-                ), If(significand >= 2, Block(
-                  twoPowE := twoPowE * 2,
-                  significand := significand / 2
-                ))),
-                // Round the significant to 23 bits of fractional part, and multiply back by twoPowE
-                absResult := roundToNearestBreakTiesToEven(significand, 1.0 / (1 << 23).toDouble) * twoPowE
+                const(p, av * 536870913),
+                absResult := (p + (av - p))
               ), {
                 // Round the value to a multiple of the smallest Float ULP, which is Float.MinPositiveValue
+                // NaN, +0 and -0 fall in this case as well, and this computation is an identity for them
```

Comment:
Just saw that: You might want to drop a comment that for the -0 case, neither `av` nor `absResult` are really absolute (as in: they end up being -0).